src/Sparrow/Platform/Win32/Win32MemoryQueryMethods.cs

```diff
@@ -24,23 +29,186 @@ struct PPSAPI_WORKING_SET_EX_INFORMATION
+            MEM_FREE = 0x10000,
+            MEM_RESERVE = 0x2000
+        }
+
+        public enum MemoryTypeConstants : uint
+        {
+            // https://msdn.microsoft.com/en-us/library/windows/desktop/aa366775(v=vs.85).aspx
+            MEM_IMAGE = 0x1000000,
+            MEM_MAPPED = 0x40000,
+            MEM_PRIVATE = 0x20000
+        }
+
+
+
+        [DllImport("kernel32.dll")]
+        public static extern void GetSystemInfo(out SYSTEM_INFO lpSystemInfo);
+
+        private static readonly byte[] VoronPostFix = new Byte[] {(byte)'.', (byte)'v', (byte)'o', (byte)'r', (byte)'o', (byte)'n'};
+        private static readonly byte[] BufferPostFix = new Byte[] { (byte)'.', (byte)'b', (byte)'u', (byte)'f', (byte)'f', (byte)'e', (byte)'r', (byte)'s' };
+
+        public static DynamicJsonArray GetMaps()
+        {
+            var dja = new DynamicJsonArray();
+
+            SYSTEM_INFO systemInfo;
+            GetSystemInfo(out systemInfo);
+
+            var procMinAddress = systemInfo.minimumApplicationAddress;
+            var procMaxAddress = systemInfo.maximumApplicationAddress;
+            var processHandle = GetCurrentProcess();
+
+            var bytesRead = new UIntPtr(0); // number of bytes read with ReadProcessMemory
+            var results = new Dictionary<string, Tuple<long, long>>();
+            var bufferArr = new byte[64 * Constants.Size.Megabyte];
+            var filenameString = new byte[2048];
+
+
+            while (procMinAddress.ToInt64() < procMaxAddress.ToInt64())
+            {
+                MEMORY_BASIC_INFORMATION memoryBasicInformation;
+                VirtualQueryEx(processHandle, (byte*)procMinAddress.ToPointer(),
+                    &memoryBasicInformation, new UIntPtr((uint)sizeof(MEMORY_BASIC_INFORMATION)));
+
+                // if this memory chunk is accessible
+                if (memoryBasicInformation.Protect == (uint)MemoryProtectionConstants.PAGE_READWRITE &&
+                    memoryBasicInformation.State == (uint)MemoryStateConstants.MEM_COMMIT &&
+                    memoryBasicInformation.Type == (uint)MemoryTypeConstants.MEM_MAPPED)
+                {
+                    var regionSize = memoryBasicInformation.RegionSize.ToInt64();
+                    long offset = 0;
+                    for (long size = 0; size < regionSize + 64 * Constants.Size.Megabyte; size += 64 * Constants.Size.Megabyte)
+                    {
+                        var partLength = size <= regionSize ? 64 * Constants.Size.Megabyte : size - regionSize;
+                        fixed (byte* buffer = bufferArr)
+                            ReadProcessMemory(processHandle,
+                                (byte*)memoryBasicInformation.BaseAddress.ToPointer() + offset, buffer, (uint)partLength, bytesRead);
+                        offset += partLength;
+
+                        var totalDirty = AddrWillCauseHardPageFault((byte*)memoryBasicInformation.BaseAddress.ToPointer(), (uint)partLength,
```

Comment:
You just _ensured_ that this address will not cause a page fault, because you read it.